docs/modules/ROOT/pages/hazelcast-platform-operator-jet-job-snapshot.adoc

```diff
@@ -0,0 +1,310 @@
+
++
+[source, shell]
+----
+cat <<EOF | kubectl apply -f -
+apiVersion: hazelcast.com/v1alpha1
+kind: JetJobSnapshot
+metadata:
+  name: snapshot-transaction
+spec:
+  name: transaction
+  jetJobResourceName: job-transaction-v1
+  cancelJob: true
+EOF
+----
+
+. Use the following command to check the status of the exported JetJobSnapshot:
+
++
+[source, shell]
+----
+$ kubectl get jetjobsnapshot snapshot-transaction
+NAME                   STATE      CREATIONTIME
+snapshot-transaction   Exported   2023-08-09T13:07:51Z
+----
+
+NOTE: By configuring the 'spec.cancelJob' field to 'true', the data pipeline named 'job-transaction-v1' will be canceled after applying the JetJobSnapshot. This setting is particularly useful before submitting a new version of the active data pipeline. With this approach, the snapshot will halt the ongoing job after preserving its current state.
+
+
+. The data pipeline should not be in the Running state anymore. You can verify this by using the following command:
+
++
+[source, shell]
+----
+$ kubectl get jetjob job-transaction-v1
+NAME                 STATUS            ID                   SUBMISSIONTIME         COMPLETIONTIME
+job-transaction-v1   ExecutionFailed   741632319877545985   2023-08-09T12:22:04Z   2023-08-09T13:07:51Z
+----
+
+== Step 4. Submit Job initialized from Snapshot
+
+When creating a new version of a data pipeline, it's essential to initialize the new pipeline from the current state of the old one. Without this initialization, the new pipeline would start with an empty state and lack information about its predecessor's state. This situation could result in data loss or duplicate processing, which is not desirable, particularly for critical pipelines. To ensure proper initialization and prevent these issues, we can rely on the Snapshot.
+
+Continuing with the example, we will now move forward to create a new version of the previous data pipeline. To maintain simplicity in the example, the new version so similar to the old one. It takes entries from the 'transaction' map and logs them.
+
+[source, java]
+----
+public static void main(String[] args) {
+    var p = Pipeline.create();
+    var transactionSource = Sources.mapJournal("transaction", JournalInitialPosition.START_FROM_OLDEST);
+    var loggerSink = Sinks.logger();
+    p.readFrom(transactionSource)
+            .withIngestionTimestamps()
+            .setName("Emit Transactions")
+            .map(e -> {
+                System.out.printf("The transaction '%s' is being executed in 'job-v2'\n", e.getKey());
+                // execute the transaction
+                return String.format("[Job V2] transaction:'%s' payload:'%s'", e.getKey(), e.getValue());
+            })
+            .setName("Apply Transactions")
+            .writeTo(loggerSink)
+            .setName("Log Transactions");
+
+    HazelcastInstance hz = Hazelcast.bootstrappedInstance();
+    hz.getJet().newJob(p);
+}
```

Comment:
WDYT about using [Dominik's approach](https://github.com/hazelcast-guides/hazelcast-platform-operator-tls/blob/master/docs/modules/ROOT/pages/hazelcast-platform-operator-tls.adoc?plain=1#L120-L127) where we use `include` functionality instead of putting the code samples hardcoded?

Reply:
It makes the doc well-structured 👍